### ISSUE TITLE
floss Crashes and gives MemoryError

### ISSUE THREAD

**Issue by oddcoder** [fireeye/flare-floss#222]
This is file I used
[re001.zip](https://github.com/fireeye/flare-floss/files/310868/re001.zip)

```
➜  ~ floss --version 
floss 1.2.0
https://github.com/fireeye/flare-floss/
➜  ~ floss diskm2.dll           

FLOSS decoded 202 strings
VA<AAA
<AAA
AAAAAAAAAAAAAAAAr
BCC`.//2
RTT0
wzz<
p044^
`ff`
Cff@Oww
8%``*
kl(xDEn
@<*,jJ{}
OR,C
Ve0,
^b"Ut:
Lugr,
$?&D
(FQP
*'?|b
2,XxKp
3T ^,
C 6rX
|OW&)
qShJ
By2l
0I$2
P%V"@
Xy~]
^+za
)@cYQ
."/0\
$zXj370
9O Vvz
~%wo^
>~N[D|~
~w_N
}jfpF
$dHiPT
tXpt
M@[5m
 b85F.Py6zj
lpe<
=AUx
2"y^
=%^p
,#FL
Z/bS
[UlJ
(30b
P7Vg
(XWR8
0HSC
h_Z7
_8bD
:S-V
VJN^+
`Oq@+
p@\a
VHPn
 `fX
5a&@
S2vrA`*
:! 5
Q)zLy|
RsW4
8`zT
0'Ld
, ~q
_6u(
xxit
y c h<
^EFt_
k vE
?x6(
6>[OX
uXnoB
qAU0
4|Vv
8%MW`1:U
w\F[Td8
 BDrC^
Gp.0?2
(#MS(
N v=hH
@/_>`
`%n&
;NIW
fu1K
812B
`8n%
r-I(
qH_7BxH1
'0c@
&Jmt2
y8j_
_<OL
VlRjT#
X':$
_HBz
2rP-
DPR8
bpkW
wkJL
$"De
TI V
fspc
`=2D
ip-`
p`bSs`K_/
@<r:
a6Onu"f
@RLj@
l~(p
urq)V
6Z|B
LT#+h
L669
o<Q[
76`X
XPJCxEW
8%yb
P #W
h,R^
D2EI
H9lc
4A9^
`,)<
vQf$
 X)N
LTz6
<k9vX+
'`xB
#I@X
Xtap
fX#?
KO@Sf
(WNE@
!8exh
]X4dS(
L t{
0Y"\P
iF`t
x` PV
6[[!
b6Pt
;VkLP^
.I[3
qpZ,
^OMD
7D+U
DRJa
df@[
VJ`X_$G
0f3~
C}>r\t#
l0d|d
=rbx
|-FV2
;@|K
@&4,
%'c 
Ui@M92
8L"`
aYh.
A }8E@
D`c?
fCl>
7`J`+";
}TzC
?!_x
UT12R
X#8Xl
8@9\t
_mVC
QX0;
]FhNB
;&[x
#^+0-N
&.~TF1
HvOrK
vpbg
8"$}
(!7P/2o0
sY0=
MZ`1d
". Y
bP;YypTV
7rEX"
_<X/
g()h
J@Hl
{K&x
rN%*IH/
Qmh`yt
Qp`|
b1pa
/oV@
p4U `d
\tfpL
"S\\t
@rfn
#7)jz
Traceback (most recent call last):
  File "/usr/bin/floss", line 9, in <module>
    load_entry_point('floss', 'console_scripts', 'floss')()
  File "/home/oddcoder/flare-floss/floss/main.py", line 675, in main
    stack_strings = list(set(stack_strings))
  File "/home/oddcoder/flare-floss/floss/stackstrings.py", line 135, in extract_stackstrings
    for s in strings.extract_ascii_strings(ctx.stack_memory):
  File "/home/oddcoder/flare-floss/floss/strings.py", line 34, in extract_ascii_strings
    if (buf[0] in REPEATS) and buf_filled_with(buf, buf[0]):
  File "/home/oddcoder/flare-floss/floss/strings.py", line 14, in buf_filled_with
    for i in range(len(buf)):
MemoryError
➜  ~ 
```

**Comment by williballenthin** [comment]
i think we're using too much memory, and the process is getting killed.

**Comment by pilate** [comment]
This one was (probably) just because of that range() in buf_filled_with, seems to run fine on that file now.

**Comment by williballenthin** [comment]
it runs, but takes up a huge amount of memory. more than 2G on my system.

this happens sometimes during stackstrings extraction:

```
~/env/bin/floss -v ~/pwd/diskm2.dll -d --no-decoded-strings --no-static-strings
```

while i'm generally ok with using available memory, 2GB seems like something might be wrong.

**Comment by williballenthin** [comment]
looks like there's a huge allocation on the stack (like 0x7e72bec7 bytes), which causes things to 1) run slowly, and 2) consume most/all available memory. need to add some better sanity checking.

**Comment by williballenthin** [comment]
fixed, assuming #228 gets merged, thanks @oddcoder!
